### ISSUE TITLE
Validation configuration is confusing

### ISSUE THREAD

**Issue by nickwp** [WatChMaL/WatChMaL#78]
During training, validation occurs multiple times within the epoch, with a subset of the validation events used for each validation. (This is unlike the standard approach of validating once at the end of each epoch, with the full validation set. The reason for doing it differently is our huge datasets resulting in large epochs)

Currently, it performs validation once every `val_interval` training iterations. Each validation consists of `num_val_batches` validation iterations.
Since both training and validation batch sizes can vary. In the current config this means the number of validations per epoch depends on the training batch size, and the proportion of the validation dataset used in each validation depends on the validation batch size.

It would be better to refactor this so that you instead explicitly set:
- The number of validations per epoch, or equivalently the number of training events between each validation
- The proportion of validation dataset to use in each validation, or equivalently the number of validation events to use in each validation

From these, it should automatically calculate:
- how many training iterations between each validation (current `val_interval` config parameter)
- how many validation iterations to perform per validation (current `num_val_batches` config parameter)

This would be nice becauseit would mean that when changing batch sizes, the validation config would not need to be changed to keep the same frequency and number of events of each validation

It would also make the config varibles hopefully more meaningful and understandable
